onnxruntime/python/ort_trainer.py

```diff
@@ -827,6 +829,12 @@ def eval_step(self, *args, **kwargs):
         input, fetches = self.prepare_input_and_fetches(self.model_desc_.inputs_,
                                                         None, None, *args, **kwargs)
 
+        if self.onnx_model_ is None:
+            if self.torch_model_ is not None:
+                self._init_onnx_model(input)
+            else:
+                raise RuntimeError("Model is unintialized. Please ensure a valid ONNX model or PyTorch model is provided to this Trainer.")
```

Comment:
I think this can be very confusing for a user, because the user may have provided a PyTorch model to the Trainer, but just didn't call the train_step. Maybe mention the need to call train_step in the error message?

Or maybe it's better to allow calling eval_step before train_step, and do the same lazy exporting here if the onnx_model_ is none?

Reply:
The coding is already doing what you are suggesting, if user provided a pytorch model to trainer, and calls eval_step before train_step, the code will export the model first.
The runtime error is that neither pytorch model nor onnx model is provided. In that case actually nothing works.

Reply:
I see now, looks good.